### PR TITLE
Implement almost all tests in GEDCOMDateTest... plus fallout

### PR DESCRIPTION
**Pull request by fire-eggs**
- Only two tests for GEDCOMDate not implemented: createObj and assign
- Nearly complete code coverage reported by JaCoCo. Non-covered code appears unreachable.
- Branch coverage misses not yet investigated.

Fallout (additional changes and bugfixes):
- Adds 'loadFromString' to GEDCOMProvider for unit testing. 
- Fixes a bug with GEDCOMFormat where gf_Ahnenblatt would never be matched.
- Added test for gf_Ahnenblatt match.

I hope I've done this right, pull requests are confusing...

### CODE REVIEW

File: projects/GKCore/test/gkcommon/gedcom/GEDCOMDateTest.java [hunk 1]
```diff
@@ -6,16 +6,29 @@
...
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
 import java.util.Date;
+import static mocks.CustomTest.assertThrows;
 import org.junit.AfterClass;
 import org.junit.BeforeClass;
 import org.junit.Test;
 import static org.junit.Assert.*;
 
+// TODO KBR date formats 20-DEC-1980,12/20/1980(american),others? createbyformattedstr() doesn't accept
```
Comment (Serg-Norseman) [inline projects/GKCore/test/gkcommon/gedcom/GEDCOMDateTest.java]: This function is used in all cases only in dialogs, in order to accept the date entered in the UI controls. Support for extended formatting for this particular function is optional.

File: projects/GKCore/test/gkcommon/gedcom/GEDCOMDateTest.java [hunk 2]
```diff
@@ -6,16 +6,29 @@
...
+// TODO KBR leap year
+// TODO KBR how does get/set datetime handle values outside the range of the Date object?
+// TODO KBR setJulian(12,20,1980) throws exception
+// TODO KBR greg <> julian conversion
+// TODO KBR use UDN to check invalid date
+
```
Comment (Serg-Norseman) [inline projects/GKCore/test/gkcommon/gedcom/GEDCOMDateTest.java]: All the changes in GEDCOMDate, please, discuss in parts.

**Comment by Serg-Norseman** [comment]
CreateObj does not need to be covered with its own test, I do not think it will significantly reduce the overall coverage. On the coverage, I will study - what can be improved.

Adding 'loadFromString' and fix for Ahnenblatt - I approve :)

Tonight or tomorrow afternoon there will be a very large commit, track the news ...

**Comment by fire-eggs** [comment]
Thanks, Sergey! 

>CreateObj does not need to be covered with its own test,
Yeah, JaCoCo said it was covered, presumably by other tests. Ditto for assign, I think.

>Tonight or tomorrow afternoon there will be a very large commit, track the news ...
Yay!

**Comment by Serg-Norseman** [comment]
Kevin, if you are not against participating in general mailing lists for the project in the future, and for more convenient communication by mail - write me a letter to the mailbox specified in about the program. And I'll answer you from my personal - to exchange addresses without their explicit publication here.

**Pull request merged**